tokens/css-utilities.js

```diff
@@ -1,21 +1,19 @@
-const chroma = require('chroma-js');
-const { darken } = require('./sass-helpers');
-
 /**
  * Implements bg-variant mixin from bootstrap. Creates utility classes for background colors based on theme color.
  */
 function bgVariant(token) {
-  const { attributes: { type, item }, value } = token;
+  const { attributes: { type, item } } = token;
   const parent = `.bg-${type}${item === 'base' ? '' : `-${item}`}`;
+  const action = token.name.replace(type, `utilities-action-default-${type}`);
```

Comment:
[question/curious] If in the future we did have `utilities-action-hover-*` and/or `utilities-action-focus-*` in the future, how would the hardcoded `default` here in `bgVariant` and `textEmphasisVariant` pull in `hover` and/or `focus` instead?

Reply:
I think that will depends on the implementation, some examples 

if  a helper/util function is created
```
function getActionVariablebyToken(token, action ="default") {
  const { attributes: { type} } = token;
  return token.name.replace(type, `utilities-action-${action}-${type}`);
}
```
if current function is modified
```
function bgVariant(token) {
  const { attributes: { type, item } } = token;
  const parent = `.bg-${type}${item === 'base' ? '' : `-${item}`}`;
  const focus = token.name.replace(type, `utilities-action-focus-${type}`);
  const hover = token.name.replace(type, `utilities-action-hover-${type}`);
  return `${parent} {
  background-color: ${`var(--${token.name})`} !important;
}

a${parent}:hover,
a${parent}:focus,
button${parent}:hover{
  background-color: ${`var(--${hover})`}  !important;
}
button${parent}:focus {
  background-color: ${`var(--${focus})`}  !important;
}

`;
}
```

Reply:
@andrey-canon Thanks for the example here. I guess my bigger concern here is that we have to hardcode a string here to match the JSON schema exactly rather than being able to infer the token name for the hover/focus color instead. For example, what if the JSON schema decides to switch "action" ➡️ "actions", we'd have to remember to make the identical change here to account for that.

That said, I'm not sure finding a solution to that is worth the lift right now unless you see an obvious way around it; I'd be OK with deferring this particular issue. Thoughts?

Reply:
@adamstankiewicz  I think a reference in the color will be necessary, something like 
```
{
  "color": {
    "primary": {
      "base": { "value": "#0A3055", "type": "color", "source": "$primary" },
      "100": {
        "value": "{color.primary.base}",
        "type": "color",
        "source": "$primary-100",
        "modify": [{ "type": "mix", "otherColor": "white", "amount": "0.94" }]
        "actions": {
          "default": "{color.action.default.primary.base}"
        }
      }
  }
}
```
and then try to get the action from the token, however, right now, I didn't have the solution  I will try to find it out